### PR TITLE
Replace TSLint rules with presets

### PR DESCRIPTION
**Pull request by nielsdB97**
As discussed in #77 I have replaced the extensive custom ruleset with some sane defaults by using the [tslint:recommended](https://github.com/palantir/tslint/blob/master/src/configs/recommended.ts) preset and added support for React by also using `tslint-react`.

Also I added `tslint-config-prettier` which disables any rule regarding code style (e.g. quote style) to avoid collisions with automatic code formatters.

### CODE REVIEW

**Comment by nielsdB97** [comment]
It would probably be nice for us to Prettify the existing template files as well, but I think that should be a separate PR.

File: packages/react-scripts/package.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     "ts-loader": "^2.3.7",
     "tsconfig-paths-webpack-plugin": "^2.0.0",
     "tslint": "^5.7.0",
+    "tslint-config-prettier": "^1.10.0",
```
Comment (wmonk) [inline packages/react-scripts/package.json]: I don't think we should add this. We should aim to be as flexible as possible.

Comment (nielsdB97) [reply packages/react-scripts/package.json]: It seemed you agreed to using this package in #77, that's why I added it. If I understand you correctly you now want to not use Prettier but instead use just the `tslint:recommended` and `tslint-react` presets?

Comment (wmonk) [reply packages/react-scripts/package.json]: Apologies, yes no style rules from prettier.

**Comment by nielsdB97** [comment]
The custom rules that you were using do not match the presets. Do you want to add custom rules to fix the linting issues or do you want to update the template files to pass the test?

**Comment by DorianGrey** [comment]
I'd favor to use `tslint-config-prettier` in the preset - even if the name tends towards `prettier`, it actually just disables any rule regarding code style (e.g. quote style) and has nothing specific for `prettier` inside. People are still free to define these rules in their particular `tslint.json`, e.g. in case they don't use an automatic code formatter like `prettier`.
Besides, that would change the linting behavior to be more similar to CRA's `eslint` setup, which also does not apply style related rules.

**Comment by wmonk** [comment]
Sorry for flip-flopping agree with @DorianGrey here, didn't realise the function of prettier in this case.

**Comment by DorianGrey** [comment]
Travis build is still failing ... which is curious, since it fails on a `.js` file.
Would you mind adding the following to the `tslint.json` to see if it makes things work?
```
"linterOptions": {
  "exclude": [
    "config/**/*.js",
    "node_modules/**/*.ts"
  ]
}
```

**Comment by DorianGrey** [comment]
LGTM, thanks!

**Pull request merged**